### PR TITLE
generic API

### PR DESCRIPTION
**Pull request by diasdavid**
> **WIP**

- spec
  - [x] id
  - [x] version
- [x] works on js-ipfs
- [x] works on js-ipfs-api

### CODE REVIEW

**Comment by diasdavid** [comment]
@nginnever any update on this one? Thank you :)

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "interface-ipfs-core",
   "version": "0.9.0",
   "description": "A test suite and interface you can use to implement a IPFS core interface.",
-  "main": "lib/index.js",
+  "main": "src/index.js",
```
Comment (diasdavid) [inline package.json]: - [ ] change this before merge and release

File: API/generic/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+generic API
```
Comment (dignifiedquire) [inline API/generic/README.md]: Calitalize

File: API/generic/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+##### `JavaScript` - ipfs.version([callback])
+
+`callback` must follow `function (err, version) {}` signature, where `err` is an error if the operation was not successful. `version` is an object with the version of the implementation, the commit and the Repo.
+
+If no `callback` is passed, a promise is returned.
+
+Example:
+
+```js
+ipfs.version(function (err, version) {
```
Comment (dignifiedquire) [inline API/generic/README.md]: What about all the options go has for version?

File: src/generic.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    describe('promise API', () => {
+      it('id', () => {
+        return ipfs.id()
+          .then((res) => {
+            expect(res).to.have.a.property('id')
+            expect(res).to.have.a.property('publicKey')
+          })
+      })
+
+      it('checks the version', () => {
```
Comment (dignifiedquire) [inline src/generic.js]: please keep the test names consistent with the cb versions

File: API/generic/README.md (hunk 3 above)
Comment (diasdavid) [inline API/generic/README.md]: They never really worked on the http-api level https://github.com/ipfs/js-ipfs-api/blob/master/test/interface-ipfs-core/version.spec.js#L8-L9, the filtering just happens on the CLI.

File: API/generic/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+#### `id`
+
+> Returns the identity of the Peer
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.id([callback])
+
+`callback` must follow `function (err, identity) {}` signature, where `err` is an error if the operation was not successful. `identity` is an object with the Peer identity.
+
```
Comment (haadcode) [inline API/generic/README.md]: Could we add an example to show what "Peer identity" means? Ie. the data that is returned.

File: API/generic/README.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+#### `version`
+
+> Returns the implementation version
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.version([callback])
+
+`callback` must follow `function (err, version) {}` signature, where `err` is an error if the operation was not successful. `version` is an object with the version of the implementation, the commit and the Repo.
+
```
Comment (haadcode) [inline API/generic/README.md]: Could we add an example of the returned `version` data?

**Pull request merged**